apis/spark/src/test/scala/io/tiledb/vcf/hail/TileDBHailVCFReaderTests.scala

```diff
@@ -0,0 +1,129 @@
+package io.tiledb.vcf.hail
+
+import java.net.URI
+import java.nio.file.Paths
+import java.util
+
+import io.tiledb.vcf.{SharedJavaSparkContext, SharedJavaSparkSession, TileDBHailVCFReader, VCFDataSourceOptions}
+import is.hail.HailContext
+import is.hail.annotations.{Region, RegionPool}
+import is.hail.backend.spark.SparkBackend
+import is.hail.expr.ir.{ExecuteContext, Interpret, MatrixIR, MatrixRead, MatrixRowsTable, TableIR, TableValue}
+import is.hail.io.fs.HadoopFS
+import is.hail.io.plink.{FamFileConfig, LoadPlink, MatrixPLINKReader, MatrixPLINKReaderParameters}
+import is.hail.io.vcf.MatrixVCFReader
+import is.hail.rvd.RVDPartitioner
+import is.hail.types.MatrixType
+import is.hail.types.virtual.{TArray, TCall, TFloat64, TLocus, TString, TStruct}
+import is.hail.utils.{ArrayBuilder, ExecutionTimer, Interval, SerializableHadoopConfiguration, TextInputFilterAndReplace, fatal, partition, using}
+import is.hail.variant.ReferenceGenome
+import org.apache.hadoop.conf.Configuration
+import org.apache.log4j.Level
+import org.apache.spark.api.java.JavaSparkContext
+import org.apache.spark.sql.catalyst.expressions.GenericRow
+import org.apache.spark.sql.sources.v2.DataSourceOptions
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.scalatest.funsuite.AnyFunSuite
+
+class TileDBHailVCFReaderTests extends AnyFunSuite {
+  val sparkSession = SparkSession
+    .builder()
+    .master("local[1]")
+    .appName("hail-integration")
+    .getOrCreate()
+
+  val sc = sparkSession.sparkContext
+  val sb = new SparkBackend("/tmp", "file:///tmp", sc)
+  val hc: HailContext = HailContext.apply(sb, "/tmp/hail.log", false, false, 50, false, 3)
+
+  val ctx = new ExecuteContext("/tmp", "file:///tmp", hc.backend,
+    new HadoopFS(new SerializableHadoopConfiguration(new Configuration)),
+    Region.apply(Region.REGULAR, RegionPool.apply(false)),
+    new ExecutionTimer("HailSuite"))
+
+  private def testTileDBVCFURI(sampleGroupName: String) = {
+    val arraysPath = Paths.get("src", "test", "resources", "arrays", "v4", sampleGroupName)
+    "file://".concat(arraysPath.toAbsolutePath.toString)
+  }
+
+  private def testVCFURI(sampleGroupName: String) = {
+    val arraysPath = Paths.get("src", "test", "resources", "inputs", sampleGroupName)
+    "file://".concat(arraysPath.toAbsolutePath.toString)
+  }
+
+  private def testSampleDataset = {
+    val dfRead = sparkSession.read.format("io.tiledb.vcf")
+      .option("uri", testTileDBVCFURI("small_vcf"))
```

Comment:
Do we need the new `small_vcf` array or can we use the existing 2 sample array? You have a filter on the sample name so the existing test array should work right?